### PR TITLE
(launch, govern) feat: support mode network

### PR DESCRIPTION
**Pull request by Tanya-atatakai**
## Proposed changes

Support mode network on Govern and Launch. Tested on virtual test nets and forks, but with lots of troubles because of limitations.

1) Launch - contract creation

https://github.com/user-attachments/assets/e1cf26fd-e5a5-48b7-8e59-8630d4bab592

2) Launch - nominated contract
image

3) Govern - Voting

https://github.com/user-attachments/assets/d8fed8d3-b36a-46b6-b956-33e9a908e0e9

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: apps/govern/components/Contracts/EditVotes/index.tsx [hunk 1]
```diff
@@ -74,7 +74,7 @@ const getColumns = (
           value={allocations[index].weight}
           status={isError ? 'error' : undefined}
           onChange={(value) => {
-            if (value) {
+            if (typeof value === 'number') {
```
Comment (Tanya-atatakai) [inline apps/govern/components/Contracts/EditVotes/index.tsx:77]: this fixes the ability to reset existing votes to zero @kupermind you complained about that it returns the value to the previous one

File: apps/launch/common-util/constants/stakingContract.ts [hunk 2]
```diff
@@ -51,6 +52,7 @@ export const IMPLEMENTATION_ADDRESSES: Addresses = {
   [base.id]: '0xEB5638eefE289691EcE01943f768EDBF96258a80',
   [arbitrum.id]: '0x04b0007b2aFb398015B76e5f22993a1fddF83644',
   [celo.id]: '0xe1E1B286EbE95b39F785d8069f2248ae9C41b7a9',
+  [mode.id]: '0xE49CB081e8d96920C38aA7AB90cb0294ab4Bc8EA',
```
Comment (Tanya-atatakai) [inline apps/launch/common-util/constants/stakingContract.ts:55]: @kupermind would be great of you could check all added addresses on this page

File: libs/util-contracts/src/lib/abiAndAddresses/stakingFactory.js [hunk 3]
```diff
@@ -10,6 +10,7 @@ export const STAKING_FACTORY = {
     [base.id]: '0x1cEe30D08943EB58EFF84DD1AB44a6ee6FEff63a',
     [arbitrum.id]: '0xEB5638eefE289691EcE01943f768EDBF96258a80',
     [celo.id]: '0x1c2cD884127b080F940b7546c1e9aaf525b1FA55',
+    [mode.id]: '0x75D529FAe220bC8db714F0202193726b46881B76',
```
Comment (Tanya-atatakai) [inline libs/util-contracts/src/lib/abiAndAddresses/stakingFactory.js:13]: @kupermind and also the factory address

File: apps/launch/common-util/constants/stakingContract.ts (hunk 2 above)
Comment (kupermind) [reply apps/launch/common-util/constants/stakingContract.ts:55]: Just FYI this is confusing for us :). We call this `StakingTokenAddress` :).

File: apps/launch/common-util/constants/stakingContract.ts [hunk 4]
```diff
@@ -81,6 +85,7 @@ export const STAKING_TOKEN_ADDRESSES: Addresses = {
   [base.id]: '0x54330d28ca3357F294334BDC454a032e7f353416',
   [arbitrum.id]: '0x3d77596beb0f130a4415df3D2D8232B3d3D31e44',
   [celo.id]: '0xaCFfAe8e57Ec6E394Eb1b41939A8CF7892DbDc51',
+  [mode.id]: '0xcfD1D50ce23C46D3Cf6407487B2F8934e96DC8f9',
```
Comment (kupermind) [inline apps/launch/common-util/constants/stakingContract.ts:88]: And we all this `olasAddress` :P

**Review by kupermind — APPROVED** [review]

**Review by DavidMinarsch — APPROVED** [review]

File: apps/launch/common-util/constants/stakingContract.ts (hunk 4 above)
Comment (Tanya-atatakai) [reply apps/launch/common-util/constants/stakingContract.ts:88]: yeah, I was also confused when updating these constants, renamed :D

**Comment by truemiller** [comment]
bit overkill to define new chains with the same chainIds as the origin fork :/
could overwrite the RPCs if on development and that would probably be enough
no biggy though

**Review by truemiller — APPROVED** [review]

**Comment by Tanya-atatakai** [comment]
> bit overkill to define new chains with the same chainIds as the origin fork :/
could overwrite the RPCs if on development and that would probably be enough
no biggy though

@truemiller good point actually, will consider changing that. but there's also something I'd like to discuss on our next eng call re this.

**Pull request merged**